packages/ceramic-cli/src/ceramic-cli-utils.ts

```diff
@@ -0,0 +1,310 @@
+            console.log(JSON.stringify(doc.content, null, 2))
+        })
+    }
+
+    /**
+     * Show document content
+     * @param docId - Document ID
+     */
+    static async show(docId: string): Promise<void> {
+        if (!CeramicCliUtils._validateDocId(docId)) {
+            console.error(`Invalid docId: ${docId}`)
+            return
+        }
+
+        await CeramicCliUtils._runWithCeramic(async (ceramic: CeramicApi) => {
+            const doc = await ceramic.loadDocument(docId)
+            console.log(JSON.stringify(doc.content, null, 2))
+        })
+    }
+
+    /**
+     * Show document state
+     * @param docId - Document ID
+     */
+    static async state(docId: string): Promise<void> {
+        if (!CeramicCliUtils._validateDocId(docId)) {
+            console.error(`Invalid docId: ${docId}`)
+            return
+        }
+
+        await CeramicCliUtils._runWithCeramic(async (ceramic: CeramicApi) => {
+            const doc = await ceramic.loadDocument(docId)
+            console.log(JSON.stringify(DoctypeUtils.serializeState(doc.state), null, 2))
+        })
+    }
+
+    /**
+     * Watch document state periodically
+     * @param docId - Document ID
+     */
+    static async watch(docId: string): Promise<void> {
+        if (!CeramicCliUtils._validateDocId(docId)) {
+            console.error(`Invalid docId: ${docId}`)
+            return
+        }
+
+        await CeramicCliUtils._runWithCeramic(async (ceramic: CeramicApi) => {
+            const doc = await ceramic.loadDocument(docId)
+            console.log(JSON.stringify(doc.content, null, 2))
+            doc.on('change', () => {
+                console.log('--- document changed ---')
+                console.log(JSON.stringify(doc.content, null, 2))
+            })
+        })
+    }
+
+    /**
+     * Get document versions
+     * @param docId - Document ID
+     */
+    static async versions(docId: string): Promise<void> {
+        if (!CeramicCliUtils._validateDocId(docId)) {
+            console.error(`Invalid docId: ${docId}`)
+            return
+        }
+
+        await CeramicCliUtils._runWithCeramic(async (ceramic: CeramicApi) => {
+            const versions = await ceramic.listVersions(docId)
+            console.log(JSON.stringify(versions, null, 2))
+        })
+    }
+
+    /**
+     * Create schema document
+     * @param content - Schema content
+     * @param owners - Schema owners
+     * @param onlyGenesis - Create only a genesis record (no publish or anchor)
+     * @param isUnique - Should document be unique?
+     */
+    static async schemaCreateDoc(content: string, owners: string, onlyGenesis: boolean, isUnique: boolean): Promise<void> {
+        try {
+            const schemaObj = JSON.parse(content)
+            if (!DoctypeUtils.isSchemaValid(schemaObj)) {
+                console.error('Invalid schema')
+                return
+            }
+        } catch (e) {
+            console.error(e)
```

Comment:
Shouldn't there be a `return` statement here? Otherwise it seems in case the `content` is not valid JSON the doc would still get created, no?

Reply:
nice catch!